### ISSUE TITLE
Travis should check that flutter/plugins uses the template where possible

### ISSUE THREAD

**Issue by collinjackson** [flutter/flutter#10108]
We could diff against the output of flutter create --plugin (with allowed exceptions for files that are normal to differ like main.dart, Info.plist, etc.) to ensure that they stay in sync with master.

This would have caught #10106 as well as a similar issue we saw in April with the hello_services sample.

**Comment by collinjackson** [comment]
@Hixie @jakobr-google @mit-mit @mravn-google @szakarias

**Comment by Hixie** [comment]
We'll never be able to do this for non-Flutter plugins, where it's even more important that we not break things. I think a better test would be to verify that a plugin written with the template as it stands today keeps working regardless of what changes we do. This is similar to what we are thinking of doing with the framework: https://github.com/flutter/flutter/issues/10004

**Comment by collinjackson** [comment]
Good point. I still think that is is useful to avoid having outdated example apps in the plugins repo but agree that #10004 would be a better solution and higher priority.

**Comment by Hixie** [comment]
Yeah we should definitely keep the plugins updated, but that's more about best practices than about avoiding errors IMHO.

**Comment by Hixie** [comment]
Not really clear how we could actually do this.

**Comment by amirh** [comment]
I think diffing the generated files might not be practical as they may be edited after generation.

One possible thing to do here is to keep a version number for project templates, and leave a trace somewhere of the template version that was used, which we could then check on CI.

**Comment by stuartmorgan** [comment]
If `flutter_migrate` is completed, we could run that regularly to update all of our examples apps.

**Comment by stuartmorgan-g** [comment]
In practice, occasionally checking in auto-migrations has worked fine for most things, and setting tooling-enforced policies for things that we really want to keep updated has covered the rest, so this isn't something I think warrants the effort of fully automating.
